packages/atomic/src/components/ipx/atomic-ipx-button/atomic-ipx-button.tsx

```diff
@@ -126,10 +128,38 @@ export class AtomicIPXButton implements InitializableComponent {
   private getIcon(icon: string) {
     const initialDiv = document.createElement('div')!;
     initialDiv.innerHTML = icon;
-    initialDiv
-      .querySelector('svg')
-      ?.setAttribute('fill', 'var(--atomic-on-primary)');
-
+    const iconElement = initialDiv.querySelector('svg');
+    if (!iconElement) {
+      return initialDiv.innerHTML; // handle null case from .querySelector
```

Comment:
nitpick: Personally find the comments superfluous and that they don't add much, since what the code does on a line by line basis is kinda straightforward anyway. But they don't really explain the initial problem/initial bug that you are trying to fix here.

What would be more explanatory about "what" the code should be fixing would be to add a cypress test for that initial bug, if possible.

Reply:
That makes a lot of sense, let me see what I can do !

Reply:
As I thought, it's really hard to track this kind of thing via cypress since it's purely visual. I'll move to a single line which gives a broader explanation and doesn't interfere with code reading as much